bin/newrelic-admin

```diff
@@ -66,6 +66,66 @@ USAGE
     system $Config{perlpath}, '-MNewFangle::Agent', @ARGV;
 }
 
+sub do_run_perl_tx {
+    GetOptions \my %opt => qw( help );
+
+    if ( $opt{help} ) {
+        print <<'USAGE';
+Usage: newrelic-admin run-perl-tx <transaction-name> ...
+
+Executes the Perl interpreter with the supplied arguments but forces
+the initialisation of the agent automatically at startup.
+
+The first argument is the transaction name used for the command being executed.
+
+If using an agent configuration file the path to the file should be
+supplied by the environment variable NEWRELIC_CONFIG_FILE. Alternatively,
+just the licence key, application and log file details can be supplied via
+environment variables NEWRELIC_LICENSE_KEY, NEWRELIC_APP_NAME and
+NEWRELIC_LOG_FILE.
+NewRelic is initialised for the host specified in NEWRELIC_DAEMON_HOST
+The transaction name specified is initited for the App specified in NEWRELIC_APP_NAME
+USAGE
+        return;
+    }
+
+    # the first argument is mandatory, it's the transaction name
+    return print_usage('run-perl-tx') unless scalar @ARGV > 1;
+    my $txn = shift @ARGV;
+
+    use NewFangle qw(newrelic_init);
+    use NewFangle::App;
+    use NewFangle::Agent;
```

Comment:
I'm not sure this will work. I'd expect loading the agent here will install the tracking code in the Perl running `newrelic-admin`, not on the child Perl that you start with `system`. You should be able to do what `run-perl` does and `-MNewFangle::Agent` before the rest of the parameters, which should DTRT.

Reply:
> the

done